classes/pickup/pickup.gd

```diff
@@ -90,11 +85,19 @@ func _pickup_id_setup() -> void:
 
 
 func _pickup_sound():
-	if sfx != null and has_node("SFXCollect"):
-		# Find an object we know will survive this object's destruction.
-		var safe_sfx_root = $"/root/Main"
-		# Anchor the sound source to that, then play it.
-		ResidualSFX.new_from_existing(sfx, safe_sfx_root)
+	# Check to make sure the object is killable.
+	if respawn_seconds == 0.0:
+		# If the has_node() check did not exist, it could pass an arg previously freed.
+		if sfx != null and has_node("SFXCollect"):
```

Comment:
I still maintain that you shouldn't check for errors that are impossible.

Instead, leave a comment _pointing out_ that the error is impossible, and preferably _explaining why._

That way, future programmers aren't worried for no reason, and they know enough to leave it safe when they make changes.

Reply:
Impossible error implies that it cannot happen under any circumstances. This error is not impossible, it's just not able to happen currently. The check is an insurance policy. I will add a comment better explaining it though.

Reply:
Indeed - including error checks to point out when code has been broken is useful, even if the code was originally bulletproof.